x-pack/plugins/cases/server/client/configure/client.ts

```diff
@@ -364,10 +407,59 @@ export async function create(
     const validatedConfigurationRequest =
       decodeWithExcessOrThrow(ConfigurationRequestRt)(configRequest);
 
-    validateDuplicatedCustomFieldKeysInRequest({
-      requestCustomFields: validatedConfigurationRequest.customFields,
+    validateDuplicatedKeysInRequest({
+      requestFields: validatedConfigurationRequest.customFields,
+      fieldName: 'customFields',
+    });
+
+    validateDuplicatedKeysInRequest({
+      requestFields: validatedConfigurationRequest.templates,
+      fieldName: 'templates',
     });
 
+    if (validatedConfigurationRequest.templates && validatedConfigurationRequest.templates.length) {
+      const hasPlatinumLicenseOrGreater = await licensingService.isAtLeastPlatinum();
+      const hasCustomFieldsInConfiguration =
+        validatedConfigurationRequest.customFields &&
+        validatedConfigurationRequest.customFields.length > 0;
+
+      validatedConfigurationRequest.templates.forEach((template, index) => {
+        if (!hasCustomFieldsInConfiguration && template.caseFields?.customFields?.length) {
```

Comment:
Isn't this checked in line 457 `validateTemplatesCustomFieldsInRequest`?

Reply:
Create and update both have similar validations for templates, I will try to move them to common function

Reply:
fixed